File: Assets/Lua/C2L/LuaCallEnum.lua
-- C# TestEnum.LOL
-- CS.命名空间.枚举名.枚举值
-- 枚举获得的是userdata自定义数据类型，获得卡语言数据类型时，就是userdata
print(CS.TestEnum.LOL)
print(CS.TestEnum.DOTA2)

-- 转换获得枚举值
print(CS.TestEnum.__CastFrom(0))
print(CS.TestEnum.__CastFrom("DOTA2"))

File: Assets/Scripts/Learn XLua/C2L/LuaCallEnum.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public enum TestEnum
{
    LOL = 0,
    DOTA2
}

public class LuaCallEnum : MonoBehaviour
{
    void Start()
    {
        XLuaEnv.Instance.DoString("require('C2L/LuaCallEnum')");
    }

    private void OnDestroy()
    {
        XLuaEnv.Instance.Free();
    }
}
